### PR TITLE
add JSON alternative support

### PR DESCRIPTION
**Pull request by UrielCh**

### CODE REVIEW

File: client.ts [hunk 1]
```diff
@@ -6,16 +6,19 @@ export interface MongoClientConstructorOptions {
   auth: AuthOptions;
   endpoint: string;
   fetch?: typeof fetch;
+  encoding?: 'JSON' | 'EJSON'
```
Comment (erfanium) [inline client.ts]:
```suggestion
  contentType?: 'ejson' | 'json';
```

it's not really about encoding.

**Review by erfanium — CHANGES_REQUESTED** [review]

**Comment by erfanium** [comment]
also you missed the `return EJSON.parse(body);` at [line](https://github.com/erfanium/atlas_sdk/blob/b9ccb49a6593f2273e4317f4b51e165d516e6566/client.ts#L230)

File: client.ts (hunk 1 above)
Comment (UrielCh) [reply client.ts]: I'm still on it...

**Comment by erfanium** [comment]
The rest is looking good to me. Thanks!

**Comment by erfanium** [comment]
@UrielCh please check this out.

> also you missed the return EJSON.parse(body); at [line](https://github.com/erfanium/atlas_sdk/blob/b9ccb49a6593f2273e4317f4b51e165d516e6566/client.ts#L230)

**Comment by UrielCh** [comment]
I had completed my change, That do not fix my issue, but I think the code is fine.

File: client.ts [hunk 2]
```diff
@@ -209,24 +223,25 @@ export class Collection<T> {
...
     const response = await this.client.fetch(url, {
       method: "POST",
       headers,
-      body: EJSON.stringify({
-        collection: this.name,
-        database: this.database.name,
-        dataSource: dataSource,
-        ...extra,
-      }),
+      body,
```
Comment (UrielCh) [inline client.ts:236]: this change was mostly for debugging and trying to write this statement in a single line, but fmt thinks that statement is still too long ... 80 char is so small...

**Comment by erfanium** [comment]
@UrielCh 
Hey!
Thanks for the PR, but I still have no idea in which real-world case, using `json` over `ejson` would be useful.

I'm not going to close this PR, I'll hit the merge button as soon as we get a few thumbs up on it. to make sure others also want this feature.
so i think you could just continue using your fork.

**Comment by UrielCh** [comment]
I'm still improving my MongoDB request-building skills.

I may or may not find an actual usage for the JSON encoder.

**Pull request closed without merging**